components/server/src/api/server.ts

```diff
@@ -126,13 +125,22 @@ export class API {
 
                     grpcServerStarted.labels(grpc_service, grpc_method, grpc_type).inc();
                     const stopTimer = grpcServerHandling.startTimer({ grpc_service, grpc_method, grpc_type });
-                    const deferred = new Deferred<ConnectError | undefined>();
-                    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-                    deferred.promise.then((err) => {
+                    const done = (err?: ConnectError) => {
                         const grpc_code = err ? Code[err.code] : "OK";
                         grpcServerHandled.labels(grpc_service, grpc_method, grpc_type, grpc_code).inc();
                         stopTimer({ grpc_code });
-                    });
+                    };
+                    const handleError = (reason: unknown) => {
+                        let err = ConnectError.from(reason, Code.Internal);
+                        if (reason != err && err.code === Code.Internal) {
+                            console.error("public api: unexpected internal error", reason);
+                            // don't leak internal errors to a user
+                            // TODO(ak) instead surface request id
+                            err = ConnectError.from(`please check server logs`, Code.Internal);
```

Comment:
yes, I want to add a request id here later that if reported we can look it up on our side, do you think it would be better if we propagate unexpected server error to the client?